src/main.ts

```diff
@@ -157,3 +165,7 @@ app.on('activate', () => {
     createWindow()
   }
 })
+
+app.on('ready', async () => {
+  global.pastelDB = await createDatabase()
```

Comment:
I don't think we need global to be honest. We can handle this with NodeJS Singleton pattern. You can create a new file, create a new object, and export it. Then you can import it, and assign "database" property on it. Next time when you import that object, it'll be pointing to this DB instance. Like this:
```js
// database.ts
export default {
  database: Database | null = null
}
```

```ts
// main.ts
import d from './database.js'
d.database = await createDatabase();
```

```ts
// anywhere
import { database } from './database.js'
// database is now our created database
```

Here is the article about it: https://medium.com/swlh/node-js-and-singleton-pattern-7b08d11c726a

Reply:
Thanks for you good solution,
Will update with this.